hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java

```diff
@@ -473,25 +483,30 @@ private void applyOrdering(TableGroup tableGroup, OrderByFragment orderByFragmen
 
 			EntityGraphTraversalState.TraversalResult traversalResult = null;
 
-			// 'entity graph' takes precedence over 'fetch profile'
-			if ( entityGraphTraversalState != null) {
-				traversalResult = entityGraphTraversalState.traverse( fetchParent, fetchable, isKeyFetchable );
-				fetchTiming = traversalResult.getFetchStrategy();
-				joined = traversalResult.isJoined();
-			}
-			else if ( loadQueryInfluencers.hasEnabledFetchProfiles() ) {
-				if ( fetchParent instanceof EntityResultGraphNode ) {
-					final EntityResultGraphNode entityFetchParent = (EntityResultGraphNode) fetchParent;
-					final EntityMappingType entityMappingType = entityFetchParent.getEntityValuedModelPart().getEntityMappingType();
-					final String fetchParentEntityName = entityMappingType.getEntityName();
-					final String fetchableRole = fetchParentEntityName + "." + fetchable.getFetchableName();
-
-					for ( String enabledFetchProfileName : loadQueryInfluencers.getEnabledFetchProfileNames() ) {
-						final FetchProfile enabledFetchProfile = creationContext.getSessionFactory().getFetchProfile( enabledFetchProfileName );
-						final org.hibernate.engine.profile.Fetch profileFetch = enabledFetchProfile.getFetchByRole( fetchableRole );
-
-						fetchTiming = FetchTiming.IMMEDIATE;
-						joined = joined || profileFetch.getStyle() == org.hibernate.engine.profile.Fetch.Style.JOIN;
+			if ( ! (fetchable instanceof CollectionPart ) ) {
```

Comment:
This is another big coding bug flushed out. We are supposed to skip entity graph traversal for collection element fetchable or map index fetchable. Traversal has been done already during the collection fetchable processing.